File: backend/entity/order.go
package entity

import (
	"gorm.io/gorm"
	"time"
)

type Orders struct {
	gorm.Model
	OrderName          string         `gorm:"uniqueIndex" json:"order_name"`
	Description        string         `json:"description"`
	OrderDate          time.Time      `json:"order_date"`
	Amount             float32           `json:"amount"`
	Service_Start_Date time.Time      `json:"service_start_date"`
	Service_End_Date   time.Time      `json:"service_end_date"`
	AddonServicesID    uint           `json:"addon_services_id"`
	AddonServices      *AddonServices `gorm:"foreignKey: addon_services_id" json:"addon_services"`
	
	//FK

	EmployerID         uint           `json:"employer_id"`
	Employer           *Employer    `gorm:"foreignKey: employer_id" json:"employer"`
}

File: frontend/src/interfaces/order.ts
import type { Addonservice } from "./addonservice";
import type { Billableitem } from "./billableitem";
import type { Employer } from "./employer";

export interface Order {
    ID: number;
    order_name: string;
    description: string;
    order_date: string;
    amount: Float32Array;
    service_start_date: Date;
    service_end_date: Date;
    add_on_service_id: number;
    add_on_service?: Addonservice;
    employer_id: number;
    employer?: Employer;
    bill_able_item_id: number;
    bill_able_item?: Billableitem;
}